pygls/server.py

```diff
@@ -147,52 +158,43 @@ def write(self, data: Any) -> None:
 
 
 class Server:
-    """Class that represents async server. It can be started using TCP or IO.
-
-    Args:
-        protocol_cls(Protocol): Protocol implementation that must be derived
-                                from `asyncio.Protocol`
+    """Base server class
 
-        converter_factory: Factory function to use when constructing a cattrs converter.
+    Parameters
+    ----------
+    protocol_cls
+       Protocol implementation that must be derive from :class:`~pygls.protocol.JsonRPCProtocol`
 
-        loop(AbstractEventLoop): asyncio event loop
+    converter_factory
+       Factory function to use when constructing a cattrs converter.
 
-        max_workers(int, optional): Number of workers for `ThreadPool` and
-                                    `ThreadPoolExecutor`
+    loop
+       The asyncio event loop
 
-        sync_kind(TextDocumentSyncKind): Text document synchronization option
-            - None(0): no synchronization
-            - Full(1): replace whole text
-            - Incremental(2): replace text within a given range
+    max_workers
+       Maximum number of workers for `ThreadPool` and `ThreadPoolExecutor`
 
-    Attributes:
-        _max_workers(int): Number of workers for thread pool executor
-        _server(Server): Server object which can be used to stop the process
-        _stop_event(Event): Event used for stopping `aio_readline`
-        _thread_pool(ThreadPool): Thread pool for executing methods decorated
-                                  with `@ls.thread()` - lazy instantiated
-        _thread_pool_executor(ThreadPoolExecutor): Thread pool executor
-                                                   passed to `run_in_executor`
-                                                    - lazy instantiated
     """
 
     def __init__(
         self,
-        protocol_cls,
-        converter_factory,
-        loop=None,
-        max_workers=2,
-        sync_kind=TextDocumentSyncKind.Incremental,
+        protocol_cls: Type[JsonRPCProtocol],
+        converter_factory: Callable[[], cattrs.Converter],
+        loop: Optional[asyncio.AbstractEventLoop] = None,
+        max_workers: int = 2,
+        sync_kind: Optional[TextDocumentSyncKind] = None,
```

Comment:
Should this remain `Incremental`?

Reply:
I think this will break servers. Many of the servers I have written assume this is set to Incremental by default.

Reply:
It *should've* been ok, since this is the base `Server` class where I don't think this was really used.
It's also set by the [LanguageServer constructor](https://github.com/openlawlibrary/pygls/pull/356/files#diff-073264baeffc39cb93e2bc9bd897133fb5d3ba00db0dd0bb073af883c15b6b61R394).

That said, I don't think we gain anything from changing this so I've rolled it back anyway :smile: